pkg/action/failback.go

```diff
@@ -79,36 +77,31 @@ func (ac *ActionController) verifyMigrationScheduleBeforeFailback(action *storkv
 	// if no migration found, abort the failback process
 	if latestMigration == nil {
 		msg := fmt.Sprintf("No migration found for migrationschedule %s, hence aborting failback", migrationSchedule.Name)
-		log.ActionLog(action).Errorf(msg)
-		ac.recorder.Event(action,
-			v1.EventTypeWarning,
-			string(storkv1.ActionStatusFailed),
-			msg)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
 		ac.updateAction(action)
 		return
 	}
 
 	if latestMigration.Status == storkv1.MigrationStatusFailed {
 		msg := fmt.Sprintf("The latest migration %s is in %s state, hence aborting failback", latestMigration.Name, storkv1.MigrationStatusFailed)
-		log.ActionLog(action).Errorf(msg)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
 		action.Status.Status = storkv1.ActionStatusFailed
 		action.Status.Reason = msg
-		ac.recorder.Event(action,
-			v1.EventTypeWarning,
-			string(storkv1.ActionStatusFailed),
-			msg)
 		ac.updateAction(action)
 		return
 	}
 
 	// Return if any migration is ongoing
 	if !isMigrationComplete(latestMigration.Status) {
 		msg := fmt.Sprintf("Waiting for the completion of migration %s", latestMigration.Name)
-		log.ActionLog(action).Infof(msg)
-		ac.recorder.Event(action,
-			v1.EventTypeWarning,
-			string(storkv1.ActionStatusScheduled),
-			msg)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
```

Comment:
Looking at the previous version it seems like ac.printFunc(action, string(storkv1.ActionStatusScheduled)) will be more appropriate.

Reply:
Done